iModelCore/ECDb/ECDb/ECSql/ECInstanceECSqlSelectAdapter.cpp

```diff
@@ -320,7 +317,7 @@ BentleyStatus ECInstanceECSqlSelectAdapter::SetPrimitiveValue(ECValueR val, ECN:
 /*---------------------------------------------------------------------------------**//**
 * @bsimethod
 +---------------+---------------+---------------+---------------+---------------+------*/
-BentleyStatus ECInstanceECSqlSelectAdapter::SetStructArrayElement(ECValueR val, ECClassCR structType, IECSqlValue const& value) const
+BentleyStatus ECInstanceECSqlSelectAdapter::SetStructElement(ECValueR val, ECClassCR structType, IECSqlValue const& value) const

```

Comment:
I'm a bit confused. You renamed the method from StructArray to just Struct, but the method contents didn't change, and it is called for struct arrays, yet it does not contain any logic regarding arrays.
Does it work with arrays or not?  Is SetStruct(nullptr) the appropriate call for a struct array?
If it all works fine, let's at least add a comment explaining that this method works for both, arrays and normal structs.

Reply:
This has been called for each of the array elements, not struct array itself. Seeing the same logic (just the non-array struct handling was lacking a null check) implemented twice, I just realized we could reuse the same SetStructArrayElement, which seems to not have anything to do with arrays inside of it.

Reply:
Resolving this, as an alternative approach has been selected.